src/index.js

```diff
@@ -126,6 +139,66 @@ class File {
   }
 }
 
+class FileInterface extends File {
+
+  get displayName() {
+    return "FileInterface";
+  }
+
+  // create and return a path url
+  get path() {
+    return  URL.createObjectURL(this.descriptor);
+  }
+
+  get encoding() {
+    return DEFAULT_ENCODING;
+  }
+
+  get size() {
+    return this.descriptor.size;
+  }
+
+  get fileName() {
+    return this.descriptor.name;
+  }
+
+  // call content function and generate MD5 hash from this content 
+  // tested with big file 1.9Gb 
+  async hash(cbProgress) {
+    return new Promise((resolve, reject) => {
+      let md5 = CryptoJS.algo.MD5.create();
```

Comment:
Just commenting for future improvements: This is going to increase the bundle size for browser, let's try use node built in `crypto` for browser instead of `CryptoJS`, since it's already being used in the proejct or use everywhere  `CryptoJS`